bundles/org.openhab.binding.yamahamusiccast/src/main/java/org/openhab/binding/yamahamusiccast/internal/YamahaMusiccastBridgeHandler.java

```diff
@@ -0,0 +1,157 @@
+                logger.debug("Received packet: {} (Tracking: {})", received, trackingID);
+                handleUDPEvent(received, trackingID);
+            } catch (SocketTimeoutException e) {
+                // Nothing to do on socket timeout
+            } catch (IOException e) {
+                logger.debug("UDP Listener got IOException waiting for datagram: {}", e.getMessage());
+                localSocket = null;
+            }
+        }
+        logger.debug("UDP Listener exiting");
+    }
+
+    public YamahaMusiccastBridgeHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+    }
+
+    @Override
+    public void initialize() {
+        updateStatus(ThingStatus.ONLINE);
+        Future<?> localEventListenerJob = eventListenerJob;
+        ExecutorService localExecutor = executor;
+        if (localEventListenerJob == null || localEventListenerJob.isCancelled()) {
+            if (localExecutor != null) {
+                localEventListenerJob = localExecutor.submit(this::receivePackets);
+            }
+        }
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        Future<?> localEventListenerJob = eventListenerJob;
+        ExecutorService localExecutor = executor;
+        if (localEventListenerJob != null) {
+            localEventListenerJob.cancel(true);
+            localEventListenerJob = null;
+        }
+        if (localExecutor != null) {
+            localExecutor.shutdownNow();
+            localExecutor = null;
+        }
+    }
+
+    public void handleUDPEvent(String json, String trackingID) {
+        String udpDeviceId = "";
+        Bridge bridge = (Bridge) thing;
+        for (Thing thing : bridge.getThings()) {
+            ThingStatusInfo statusInfo = thing.getStatusInfo();
+            switch (statusInfo.getStatus()) {
+                case ONLINE:
+                    logger.debug("Thing Status: ONLINE - {}", thing.getLabel());
+                    YamahaMusiccastHandler handler = (YamahaMusiccastHandler) thing.getHandler();
+                    if (handler != null) {
+                        logger.debug("UDP: {} - {} ({} - Tracking: {})", json, handler.getDeviceId(), thing.getLabel(),
+                                trackingID);
+
+                        @Nullable
+                        UdpMessage targetObject = gson.fromJson(json, UdpMessage.class);
+                        if (targetObject != null) {
+                            udpDeviceId = targetObject.getDeviceId();
+                            if (udpDeviceId.equals(handler.getDeviceId())) {
+                                handler.processUDPEvent(json, trackingID);
+                            }
+                        }
+                    }
+                    break;
+                default:
+                    logger.debug("Thing Status: NOT ONLINE - {} (Tracking: {})", thing.getLabel(), trackingID);
```

Comment:
See above. Logging.

Reply:
Set all _debug_ to _trace_ as this is more about verbose logging to give user insight in what is happening. At the same time, that information in necessary to solve problems.